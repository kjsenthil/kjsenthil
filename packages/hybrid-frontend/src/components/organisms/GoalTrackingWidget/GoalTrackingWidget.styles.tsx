import styled, { keyframes, css } from 'styled-components';
import { CardContent } from '../../atoms';

export const ProgressLabelContainer = styled.div`
  position: relative;
  flex-direction: row;
  width: 100%;
  overflow: hidden;
`;

export const ProgressLabelWrapper = styled.div`
  ${({ width }: { width: number }) => {
    const fillAnimation = keyframes`
      from {
        width: 0;
      }
      to {
        width: ${width}%;
      }
    `;

    return css`
      min-width: 90px;
      max-width: 50%;
      width: ${width}%;
      animation: ${fillAnimation} 1s ease-out;
      z-index: 1;
    `;
  }}
`;

export const GoalTrackingCardContent = styled(CardContent)`
  &.MuiCardContent-root:last-child {
    padding-bottom: 0;
  }
`;
