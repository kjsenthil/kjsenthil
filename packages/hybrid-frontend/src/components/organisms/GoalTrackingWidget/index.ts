import { GoalTrackingWidgetProps } from './GoalTrackingWidget';

export { default } from './GoalTrackingWidget';
export type { GoalTrackingWidgetProps };
