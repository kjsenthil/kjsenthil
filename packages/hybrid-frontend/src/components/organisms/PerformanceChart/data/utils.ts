export interface GetPerformanceDataResponse {
  // All numbers have 2 decimal places
  Data: {
    Type: 'performance';
    Id: string;
    Attributes: {
      AccountValue: number;
      Performance: {
        Value: number;
        Percentage: number; // This is a float. 50% will be 50.00
      };
      Values: Array<{ Date: string; Value: number }>;
    };
    Links: {
      self: string;
    };
    Relationships: {
      contributions: {
        Links: {
          related: string;
        };
        Data: {
          Type: 'contributions';
          Id: string;
        };
      };
    };
  };
  Included: Array<{
    // All numbers have 3 decimal places
    Type: 'contributions';
    Id: string;
    Attributes: {
      TotalContributions: number;
      Contributions: Array<{ Date: string; NetContributionsToDate: number }>;
    };
    Links: {
      self: string;
    };
    Relationships: null;
  }>;
}

export interface TimeSeriesDatum {
  date: Date;
  value: number;
}

export interface PerformanceDatum extends TimeSeriesDatum {
  metadata?: Record<string, unknown>;
}

export interface ContributionDatum extends TimeSeriesDatum {
  metadata?: Record<string, unknown>;
}

export enum PerformanceChartPeriod {
  '1M' = '1m',
  '3M' = '3m',
  '6M' = '6m',
  '1Y' = '1y',
  'ALL_TIME' = 'alltime',
}

export function sortTimeSeries<TS extends TimeSeriesDatum>(
  timeSeriesData: TS[],
  direction: 'asc' | 'desc' = 'asc'
): TS[] {
  return [...timeSeriesData].sort((d1, d2) =>
    direction === 'asc'
      ? d2.date.getTime() - d1.date.getTime()
      : d1.date.getTime() - d2.date.getTime()
  );
}

export function getTimeSeriesMinMax<TS extends TimeSeriesDatum>(timeSeriesData: TS[]) {
  if (timeSeriesData.length === 0) {
    return {
      minDate: new Date(0),
      maxDate: new Date(0),
      minValue: 0,
      maxValue: 0,
    };
  }

  let minDate = timeSeriesData[0].date;
  let maxDate = timeSeriesData[0].date;
  let minValue = timeSeriesData[0].value;
  let maxValue = timeSeriesData[0].value;

  timeSeriesData.forEach(({ date, value }) => {
    if (date > maxDate) maxDate = date;
    if (date < minDate) minDate = date;
    if (value > maxValue) maxValue = value;
    if (value < minValue) minValue = value;
  });

  return {
    minDate,
    maxDate,
    minValue,
    maxValue,
  };
}

export function timeSeriesDateAccessor(d: TimeSeriesDatum) {
  return d.date;
}

export function timeSeriesValueAccessor(d: TimeSeriesDatum) {
  return d.value;
}
