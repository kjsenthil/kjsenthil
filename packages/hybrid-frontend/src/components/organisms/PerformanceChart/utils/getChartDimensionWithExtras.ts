import { ChartDimension, ChartDimensionWithExtras } from '../../../../config/chart';

/**
 * This function calculates and injects certain extra figures (like inner width
 * and inner height) to a chart dimension object. It returns a new chart
 * dimension object containing these extras.
 */
export default function getChartDimensionWithExtras(
  chartDimension: ChartDimension
): ChartDimensionWithExtras {
  return {
    ...chartDimension,
    innerWidth: chartDimension.width - chartDimension.margin.left - chartDimension.margin.right,
    innerHeight: chartDimension.height - chartDimension.margin.top - chartDimension.margin.bottom,
  };
}
