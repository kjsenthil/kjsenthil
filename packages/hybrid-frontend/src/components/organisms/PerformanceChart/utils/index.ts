/* eslint-disable import/prefer-default-export */
export { default as getChartDimensionWithExtras } from './getChartDimensionWithExtras';
