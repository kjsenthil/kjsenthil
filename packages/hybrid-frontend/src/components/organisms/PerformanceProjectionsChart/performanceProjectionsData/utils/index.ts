export { default as valueDefinedFactory } from './contributionsDefined';
export { default as mapDate } from './mapDate';

export * from './accessors';
export * from './minMax';
