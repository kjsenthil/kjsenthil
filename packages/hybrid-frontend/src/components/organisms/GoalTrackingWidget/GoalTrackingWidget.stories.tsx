import * as React from 'react';
import { Meta, Story } from '@storybook/react/types-6-0';
import GoalTrackingWidget, { GoalTrackingWidgetProps } from './GoalTrackingWidget';
import { formatCurrency, formatPercent } from '../../../utils/formatters';

export default {
  title: 'Digital Hybrid/Organisms/Goal Tracking Widget',
  component: GoalTrackingWidget,
} as Meta;

const Template: Story<GoalTrackingWidgetProps> = (args) => <GoalTrackingWidget {...args} />;

const progressBarData = [
  {
    progress: 0.2,
    legendProps: { title: 'Lump Sum', value: 20000 },
  },
  {
    progress: 0.5,
    legendProps: { title: 'From Age 67 - 89', value: 1080000 },
  },
  {
    progress: 0.05,
    legendProps: { title: 'Remaining', value: 72000 },
  },
];

const opts = { minimumFractionDigits: 0 };
const onTrack = formatPercent(0.72, { opts });
const shortfall = 55000;
const totalProjected = progressBarData.reduce((acc, data) => acc + data.legendProps.value, 0);
const target = totalProjected + shortfall;

export const Default = Template.bind({});
Default.args = {
  currencyFormatter: formatCurrency,
  target,
  onTrack,
  totalProjected,
  progressBarData,
  drawdownMonthlyIncome: 2340,
  drawdownStartAge: 67,
  drawdownEndAge: 89,
};
