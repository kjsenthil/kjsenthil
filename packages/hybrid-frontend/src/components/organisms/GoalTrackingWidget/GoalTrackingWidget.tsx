import * as React from 'react';
import { formatPercent, formatCurrency } from '../../../utils/formatters';
import {
  Grid,
  Typography,
  Spacer,
  Card,
  Icon,
  Divider,
  useTheme,
  useMediaQuery,
} from '../../atoms';
import { TypographyWithTooltip } from '../../molecules';
import ProgressBarWithLegend, {
  ProgressBarWithLegendProps,
} from '../../molecules/ProgressBarWithLegend';
import { GoalTrackingCardContent } from './GoalTrackingWidget.styles';

export interface GoalTrackingWidgetProps
  extends Omit<ProgressBarWithLegendProps, 'currencyFormatter'> {
  target: number;
  onTrack: number;
  totalProjected: number;
  drawdownStartAge: number;
  surplusOrShortfall: number;
  drawdownEndAge: number;
  drawdownMonthlyIncome: number;
}

const GoalTrackingWidget = ({
  target,
  totalProjected,
  onTrack,
  drawdownStartAge,
  surplusOrShortfall,
  drawdownEndAge,
  drawdownMonthlyIncome,
  progressBarData,
}: GoalTrackingWidgetProps) => {
  const theme = useTheme();
  const isMobile = useMediaQuery(theme.breakpoints.down('xs' as any));

  return (
    <Card variant="outlined">
      <GoalTrackingCardContent>
        <Grid container direction="column" spacing={0}>
          <Grid item container spacing={2} wrap="nowrap" justify="flex-start" xs={12}>
            {isMobile || (
              <Grid item>
                <Icon name="infoCircleIcon" color="secondary" />
              </Grid>
            )}
            <Grid item container direction="column" xs={12}>
              <Grid item direction="row" container alignItems="center">
                <Spacer x={1} />
                <Typography color="primary" colorShade="dark2" variant="h3">
                  {formatCurrency(totalProjected)}
                </Typography>
                <span>{'\u00a0\u00a0/\u00a0\u00a0'}</span>
                <Typography color="grey" colorShade="dark1" variant="sh3">
                  {formatCurrency(target)}
                </Typography>
              </Grid>
              <Spacer y={2} />
              <Grid item container direction="row" alignItems="center">
                <Spacer x={1} />
                <TypographyWithTooltip
                  tooltip="Some description"
                  typographyProps={{ color: 'grey', colorShade: 'dark1' }}
                >
                  You&#39;re on track to have{' '}
                  {formatPercent(onTrack, { opts: { minimumFractionDigits: 0 } })} of your target.
                  That&#39;s a {onTrack > 1 ? `surplus` : `shortfall`} of
                  {'\u00a0'}
                  {formatCurrency(Math.abs(surplusOrShortfall))}.
                </TypographyWithTooltip>
              </Grid>
              <Spacer y={2} />
              <ProgressBarWithLegend
                progressBarData={progressBarData}
                currencyFormatter={formatCurrency}
              />
            </Grid>
          </Grid>
        </Grid>
        <Grid item xs={12}>
          <Spacer y={2} />
          <Divider />
          <Spacer y={2} />
        </Grid>
        <Grid item xs={12}>
          <Grid container direction="row" wrap="nowrap" spacing={2} justify="flex-start">
            {isMobile || (
              <Grid item>
                <Spacer x={3} />
              </Grid>
            )}
            <Grid item container direction="column" xs={12}>
              <TypographyWithTooltip
                tooltip="Some description"
                typographyProps={{ color: 'grey', colorShade: 'dark1' }}
              >
                You&#39;re likely to have {formatCurrency(drawdownMonthlyIncome)} to spend each
                month between ages {drawdownStartAge} - {drawdownEndAge}.
              </TypographyWithTooltip>
              <Spacer y={2} />
            </Grid>
          </Grid>
        </Grid>
      </GoalTrackingCardContent>
    </Card>
  );
};

export default GoalTrackingWidget;
