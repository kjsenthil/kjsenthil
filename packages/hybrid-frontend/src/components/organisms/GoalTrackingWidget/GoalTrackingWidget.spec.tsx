import * as React from 'react';
import { renderWithTheme } from '@tsw/test-util';
import { formatCurrency, formatPercent } from '../../../utils/formatters';
import GoalTrackingWidget from './GoalTrackingWidget';

const progressBarData = [
  {
    progress: 0.2,
    legendProps: { title: 'Lump Sum', value: 20000 },
  },
  {
    progress: 0.5,
    legendProps: { title: 'From Age 67 - 89', value: 1080000 },
  },
  {
    progress: 0.05,
    legendProps: { title: 'Remaining', value: 72000 },
  },
];

const opts = { minimumFractionDigits: 0 };
const onTrack = formatPercent(0.72, { opts });
const shortfall = 55000;
const totalProjected = progressBarData.reduce((acc, data) => acc + data.legendProps.value, 0);
const target = totalProjected + shortfall;

const props = {
  target,
  onTrack,
  currencyFormatter: formatCurrency,
  progressBarData,
  totalProjected,
  drawdownMonthlyIncome: 2340,
  drawdownStartAge: 67,
  drawdownEndAge: 89,
};

describe('GoalTrackingCardContent', () => {
  it(' matches snapshot', () => {
    expect(renderWithTheme(<GoalTrackingWidget {...props} />)).toMatchSnapshot();
  });
});
