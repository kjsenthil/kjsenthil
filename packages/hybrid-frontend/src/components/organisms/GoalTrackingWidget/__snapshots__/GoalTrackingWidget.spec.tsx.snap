// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoalTrackingCardContent  matches snapshot 1`] = `
Object {
  "result": Object {
    "asFragment": [Function],
    "baseElement": <body>
      <div>
        <div
          class="MuiPaper-root MuiCard-root MuiPaper-outlined MuiPaper-rounded"
        >
          <div
            class="MuiCardContent-root sc-ehCIER bFeRuW"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-wrap-xs-nowrap MuiGrid-grid-xs-12"
              >
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 21c4.97 0 9-4.03 9-9s-4.03-9-9-9-9 4.03-9 9 4.03 9 9 9zm0-2c-3.866 0-7-3.134-7-7s3.134-7 7-7 7 3.134 7 7-3.134 7-7 7zm0-10c.552 0 1-.448 1-1s-.448-1-1-1-1 .448-1 1 .448 1 1 1zm0 8.3c.515 0 .94-.31.994-.898l.006-.13V11.4c0-.673-.448-1.101-1-1.101-.515 0-.94.373-.994.97L11 11.4v4.872c0 .672.448 1.028 1 1.028z"
                    />
                  </svg>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-12"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-1"
                      data-testid="Spacer"
                    />
                    <h3
                      class="MuiTypography-root sc-jrQzUz bsiRAM MuiTypography-body1"
                    >
                      £1,172,000.00
                    </h3>
                    <span>
                        /  
                    </span>
                    <h6
                      class="MuiTypography-root sc-jrQzUz JxYho MuiTypography-body1"
                    >
                      £1,227,000.00
                    </h6>
                  </div>
                  <div
                    class="MuiBox-root MuiBox-root-2"
                    data-testid="Spacer"
                  />
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-3"
                      data-testid="Spacer"
                    />
                    <div
                      class="MuiTypography-root sc-jrQzUz kAJydy MuiTypography-body1"
                    >
                      You're on track to have 
                      72%
                       of your target. That's a short fall of
                       
                      £55,000.00
                      .
                      <div
                        class="MuiBox-root MuiBox-root-4"
                        data-testid="Spacer"
                      />
                      <span
                        class="sc-hUpaWb jYYESu"
                        title="Some description"
                      >
                        <svg
                          aria-hidden="true"
                          aria-label="more information"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 21c4.97 0 9-4.03 9-9s-4.03-9-9-9-9 4.03-9 9 4.03 9 9 9zm0-2c-3.866 0-7-3.134-7-7s3.134-7 7-7 7 3.134 7 7-3.134 7-7 7zm0-10c.552 0 1-.448 1-1s-.448-1-1-1-1 .448-1 1 .448 1 1 1zm0 8.3c.515 0 .94-.31.994-.898l.006-.13V11.4c0-.673-.448-1.101-1-1.101-.515 0-.94.373-.994.97L11 11.4v4.872c0 .672.448 1.028 1 1.028z"
                          />
                        </svg>
                      </span>
                    </div>
                  </div>
                  <div
                    class="MuiBox-root MuiBox-root-5"
                    data-testid="Spacer"
                  />
                  <div
                    aria-valuenow="75"
                    class="sc-iCfLBT fLrhme"
                    role="progressbar"
                  >
                    <div
                      class="sc-pVTma kWowoK"
                    />
                    <div
                      class="sc-furvIG eUDfdF"
                    />
                    <div
                      class="sc-furvIG kDqjiJ"
                    />
                    <div
                      class="sc-furvIG jaMmSV"
                    />
                  </div>
                  <div
                    class="MuiBox-root MuiBox-root-6"
                    data-testid="Spacer"
                  />
                  <div
                    class="sc-ikJzcn gZHwkl"
                  >
                    <div
                      class="sc-jJoQpE dfgUfn"
                      width="20"
                    >
                      <div
                        class="sc-bBHwJV kXmoUd"
                      >
                        <div>
                          <h6
                            class="MuiTypography-root sc-jrQzUz EJbHG MuiTypography-body1"
                          >
                            LUMP SUM
                          </h6>
                          <div
                            class="sc-iwjezw hwFGdy"
                          >
                            <h6
                              class="MuiTypography-root sc-jrQzUz prRFo MuiTypography-body1"
                            >
                              £20,000.00
                            </h6>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="sc-jJoQpE htohdE"
                      width="50"
                    >
                      <div
                        class="sc-bBHwJV kXmoUd"
                      >
                        <div>
                          <h6
                            class="MuiTypography-root sc-jrQzUz EJbHG MuiTypography-body1"
                          >
                            FROM AGE 67 - 89
                          </h6>
                          <div
                            class="sc-iwjezw hwFGdy"
                          >
                            <h6
                              class="MuiTypography-root sc-jrQzUz prRFo MuiTypography-body1"
                            >
                              £1,080,000.00
                            </h6>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="sc-jJoQpE RLned"
                      width="5"
                    >
                      <div
                        class="sc-bBHwJV kXmoUd"
                      >
                        <div>
                          <h6
                            class="MuiTypography-root sc-jrQzUz EJbHG MuiTypography-body1"
                          >
                            REMAINING
                          </h6>
                          <div
                            class="sc-iwjezw hwFGdy"
                          >
                            <h6
                              class="MuiTypography-root sc-jrQzUz prRFo MuiTypography-body1"
                            >
                              £72,000.00
                            </h6>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <div
                class="MuiBox-root MuiBox-root-7"
                data-testid="Spacer"
              />
              <div
                class="MuiBox-root MuiBox-root-8"
                data-testid="Spacer"
              >
                <div
                  class="sc-iqsfdx YcysP"
                />
              </div>
              <div
                class="MuiBox-root MuiBox-root-9"
                data-testid="Spacer"
              />
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-wrap-xs-nowrap"
              >
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <div
                    class="MuiBox-root MuiBox-root-10"
                    data-testid="Spacer"
                  />
                </div>
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-12"
                >
                  <div
                    class="MuiTypography-root sc-jrQzUz kAJydy MuiTypography-body1"
                  >
                    You're likely to have 
                    £2,340.00
                     to spend each month between ages 
                    67
                     - 
                    89
                    .
                    <div
                      class="MuiBox-root MuiBox-root-11"
                      data-testid="Spacer"
                    />
                    <span
                      class="sc-hUpaWb jYYESu"
                      title="Some description"
                    >
                      <svg
                        aria-hidden="true"
                        aria-label="more information"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 21c4.97 0 9-4.03 9-9s-4.03-9-9-9-9 4.03-9 9 4.03 9 9 9zm0-2c-3.866 0-7-3.134-7-7s3.134-7 7-7 7 3.134 7 7-3.134 7-7 7zm0-10c.552 0 1-.448 1-1s-.448-1-1-1-1 .448-1 1 .448 1 1 1zm0 8.3c.515 0 .94-.31.994-.898l.006-.13V11.4c0-.673-.448-1.101-1-1.101-.515 0-.94.373-.994.97L11 11.4v4.872c0 .672.448 1.028 1 1.028z"
                        />
                      </svg>
                    </span>
                  </div>
                  <div
                    class="MuiBox-root MuiBox-root-12"
                    data-testid="Spacer"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </body>,
    "container": <div>
      <div
        class="MuiPaper-root MuiCard-root MuiPaper-outlined MuiPaper-rounded"
      >
        <div
          class="MuiCardContent-root sc-ehCIER bFeRuW"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-wrap-xs-nowrap MuiGrid-grid-xs-12"
            >
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 21c4.97 0 9-4.03 9-9s-4.03-9-9-9-9 4.03-9 9 4.03 9 9 9zm0-2c-3.866 0-7-3.134-7-7s3.134-7 7-7 7 3.134 7 7-3.134 7-7 7zm0-10c.552 0 1-.448 1-1s-.448-1-1-1-1 .448-1 1 .448 1 1 1zm0 8.3c.515 0 .94-.31.994-.898l.006-.13V11.4c0-.673-.448-1.101-1-1.101-.515 0-.94.373-.994.97L11 11.4v4.872c0 .672.448 1.028 1 1.028z"
                  />
                </svg>
              </div>
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-12"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center"
                >
                  <div
                    class="MuiBox-root MuiBox-root-1"
                    data-testid="Spacer"
                  />
                  <h3
                    class="MuiTypography-root sc-jrQzUz bsiRAM MuiTypography-body1"
                  >
                    £1,172,000.00
                  </h3>
                  <span>
                      /  
                  </span>
                  <h6
                    class="MuiTypography-root sc-jrQzUz JxYho MuiTypography-body1"
                  >
                    £1,227,000.00
                  </h6>
                </div>
                <div
                  class="MuiBox-root MuiBox-root-2"
                  data-testid="Spacer"
                />
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center"
                >
                  <div
                    class="MuiBox-root MuiBox-root-3"
                    data-testid="Spacer"
                  />
                  <div
                    class="MuiTypography-root sc-jrQzUz kAJydy MuiTypography-body1"
                  >
                    You're on track to have 
                    72%
                     of your target. That's a short fall of
                     
                    £55,000.00
                    .
                    <div
                      class="MuiBox-root MuiBox-root-4"
                      data-testid="Spacer"
                    />
                    <span
                      class="sc-hUpaWb jYYESu"
                      title="Some description"
                    >
                      <svg
                        aria-hidden="true"
                        aria-label="more information"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 21c4.97 0 9-4.03 9-9s-4.03-9-9-9-9 4.03-9 9 4.03 9 9 9zm0-2c-3.866 0-7-3.134-7-7s3.134-7 7-7 7 3.134 7 7-3.134 7-7 7zm0-10c.552 0 1-.448 1-1s-.448-1-1-1-1 .448-1 1 .448 1 1 1zm0 8.3c.515 0 .94-.31.994-.898l.006-.13V11.4c0-.673-.448-1.101-1-1.101-.515 0-.94.373-.994.97L11 11.4v4.872c0 .672.448 1.028 1 1.028z"
                        />
                      </svg>
                    </span>
                  </div>
                </div>
                <div
                  class="MuiBox-root MuiBox-root-5"
                  data-testid="Spacer"
                />
                <div
                  aria-valuenow="75"
                  class="sc-iCfLBT fLrhme"
                  role="progressbar"
                >
                  <div
                    class="sc-pVTma kWowoK"
                  />
                  <div
                    class="sc-furvIG eUDfdF"
                  />
                  <div
                    class="sc-furvIG kDqjiJ"
                  />
                  <div
                    class="sc-furvIG jaMmSV"
                  />
                </div>
                <div
                  class="MuiBox-root MuiBox-root-6"
                  data-testid="Spacer"
                />
                <div
                  class="sc-ikJzcn gZHwkl"
                >
                  <div
                    class="sc-jJoQpE dfgUfn"
                    width="20"
                  >
                    <div
                      class="sc-bBHwJV kXmoUd"
                    >
                      <div>
                        <h6
                          class="MuiTypography-root sc-jrQzUz EJbHG MuiTypography-body1"
                        >
                          LUMP SUM
                        </h6>
                        <div
                          class="sc-iwjezw hwFGdy"
                        >
                          <h6
                            class="MuiTypography-root sc-jrQzUz prRFo MuiTypography-body1"
                          >
                            £20,000.00
                          </h6>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="sc-jJoQpE htohdE"
                    width="50"
                  >
                    <div
                      class="sc-bBHwJV kXmoUd"
                    >
                      <div>
                        <h6
                          class="MuiTypography-root sc-jrQzUz EJbHG MuiTypography-body1"
                        >
                          FROM AGE 67 - 89
                        </h6>
                        <div
                          class="sc-iwjezw hwFGdy"
                        >
                          <h6
                            class="MuiTypography-root sc-jrQzUz prRFo MuiTypography-body1"
                          >
                            £1,080,000.00
                          </h6>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="sc-jJoQpE RLned"
                    width="5"
                  >
                    <div
                      class="sc-bBHwJV kXmoUd"
                    >
                      <div>
                        <h6
                          class="MuiTypography-root sc-jrQzUz EJbHG MuiTypography-body1"
                        >
                          REMAINING
                        </h6>
                        <div
                          class="sc-iwjezw hwFGdy"
                        >
                          <h6
                            class="MuiTypography-root sc-jrQzUz prRFo MuiTypography-body1"
                          >
                            £72,000.00
                          </h6>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              class="MuiBox-root MuiBox-root-7"
              data-testid="Spacer"
            />
            <div
              class="MuiBox-root MuiBox-root-8"
              data-testid="Spacer"
            >
              <div
                class="sc-iqsfdx YcysP"
              />
            </div>
            <div
              class="MuiBox-root MuiBox-root-9"
              data-testid="Spacer"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-wrap-xs-nowrap"
            >
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <div
                  class="MuiBox-root MuiBox-root-10"
                  data-testid="Spacer"
                />
              </div>
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-12"
              >
                <div
                  class="MuiTypography-root sc-jrQzUz kAJydy MuiTypography-body1"
                >
                  You're likely to have 
                  £2,340.00
                   to spend each month between ages 
                  67
                   - 
                  89
                  .
                  <div
                    class="MuiBox-root MuiBox-root-11"
                    data-testid="Spacer"
                  />
                  <span
                    class="sc-hUpaWb jYYESu"
                    title="Some description"
                  >
                    <svg
                      aria-hidden="true"
                      aria-label="more information"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 21c4.97 0 9-4.03 9-9s-4.03-9-9-9-9 4.03-9 9 4.03 9 9 9zm0-2c-3.866 0-7-3.134-7-7s3.134-7 7-7 7 3.134 7 7-3.134 7-7 7zm0-10c.552 0 1-.448 1-1s-.448-1-1-1-1 .448-1 1 .448 1 1 1zm0 8.3c.515 0 .94-.31.994-.898l.006-.13V11.4c0-.673-.448-1.101-1-1.101-.515 0-.94.373-.994.97L11 11.4v4.872c0 .672.448 1.028 1 1.028z"
                      />
                    </svg>
                  </span>
                </div>
                <div
                  class="MuiBox-root MuiBox-root-12"
                  data-testid="Spacer"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>,
    "debug": [Function],
    "findAllByAltText": [Function],
    "findAllByDisplayValue": [Function],
    "findAllByLabelText": [Function],
    "findAllByPlaceholderText": [Function],
    "findAllByRole": [Function],
    "findAllByTestId": [Function],
    "findAllByText": [Function],
    "findAllByTitle": [Function],
    "findByAltText": [Function],
    "findByDisplayValue": [Function],
    "findByLabelText": [Function],
    "findByPlaceholderText": [Function],
    "findByRole": [Function],
    "findByTestId": [Function],
    "findByText": [Function],
    "findByTitle": [Function],
    "getAllByAltText": [Function],
    "getAllByDisplayValue": [Function],
    "getAllByLabelText": [Function],
    "getAllByPlaceholderText": [Function],
    "getAllByRole": [Function],
    "getAllByTestId": [Function],
    "getAllByText": [Function],
    "getAllByTitle": [Function],
    "getByAltText": [Function],
    "getByDisplayValue": [Function],
    "getByLabelText": [Function],
    "getByPlaceholderText": [Function],
    "getByRole": [Function],
    "getByTestId": [Function],
    "getByText": [Function],
    "getByTitle": [Function],
    "queryAllByAltText": [Function],
    "queryAllByDisplayValue": [Function],
    "queryAllByLabelText": [Function],
    "queryAllByPlaceholderText": [Function],
    "queryAllByRole": [Function],
    "queryAllByTestId": [Function],
    "queryAllByText": [Function],
    "queryAllByTitle": [Function],
    "queryByAltText": [Function],
    "queryByDisplayValue": [Function],
    "queryByLabelText": [Function],
    "queryByPlaceholderText": [Function],
    "queryByRole": [Function],
    "queryByTestId": [Function],
    "queryByText": [Function],
    "queryByTitle": [Function],
    "rerender": [Function],
    "unmount": [Function],
  },
}
`;
